include/boost/json/string.hpp

```diff
@@ -2782,7 +2792,7 @@ typename std::enable_if<
 operator==(T const& lhs, U const& rhs) noexcept
 #endif
 {
-    return string_view(lhs) == string_view(rhs);
+    return lhs.subview() == rhs.subview();
```

Comment:
This may not work if either `T` or `U` is neither `string` or `string_view`. You're testing for `is_convertible`, which allows any type that may not have `subview()` member.